cmd/cluster-displayer/main.go

```diff
@@ -0,0 +1,199 @@
+	}
+	var page Page
+	page.Data = []map[string]string{}
+	for _, p := range clusterPools.Items {
+		maxSize := "nil"
+		if p.Spec.MaxSize != nil {
+			maxSize = strconv.FormatInt(int64(*p.Spec.MaxSize), 10)
+		}
+		page.Data = append(
+			page.Data, map[string]string{
+				"namespace": p.Namespace,
+				"name":      p.Name,
+				"ready":     strconv.FormatInt(int64(p.Status.Ready), 10),
+				"size":      strconv.FormatInt(int64(p.Spec.Size), 10),
+				"maxSize":   maxSize,
+				"imageSet":  p.Spec.ImageSetRef.Name,
+				"labels":    getLabelsString(p.Labels),
+			},
+		)
+	}
+	return &page, nil
+}
+
+func getResource(ctx context.Context, hiveClient ctrlruntimeclient.Client, resource string) (*Page, error) {
+	switch strings.ToLower(resource) {
+	case "clusterpools":
+		return getClusterPoolPage(ctx, hiveClient)
+	default:
+		return nil, fmt.Errorf("unknown resource: %s", resource)
+	}
+}
+
+func getRouter(ctx context.Context, hiveClient ctrlruntimeclient.Client) *mux.Router {
+	handler := mux.NewRouter()
+
+	handler.HandleFunc("/api/health", func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Content-Type", "application/json")
+		json.NewEncoder(w).Encode(map[string]bool{"ok": true})
+	}).Methods(http.MethodGet)
+
+	handler.HandleFunc("/api/v1/{resource}", func(w http.ResponseWriter, r *http.Request) {
+		vars := mux.Vars(r)
+		resource := vars["resource"]
+		logrus.WithField("resource", resource).Info("serving resource")
+		page, err := getResource(ctx, hiveClient, resource)
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+		}
+		callbackName := r.URL.Query().Get("callback")
+		bytes, err := json.Marshal(page)
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+		}
+		w.Header().Set("Content-Type", "application/javascript")
+		fmt.Fprintf(w, "%s(%s);", callbackName, string(bytes))
+	}).Methods(http.MethodGet).Queries("callback", "{callback}")
+
+	handler.HandleFunc("/api/v1/{resource}", func(w http.ResponseWriter, r *http.Request) {
+		vars := mux.Vars(r)
+		resource := vars["resource"]
+		logrus.WithField("resource", resource).Info("serving resource")
+		page, err := getResource(ctx, hiveClient, resource)
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+		}
+		w.Header().Set("Content-Type", "application/json")
+		json.NewEncoder(w).Encode(page)
+	}).Methods(http.MethodGet)
+	return handler
+}
+
+func main() {
+	logrusutil.ComponentInit()
+	o, err := gatherOptions()
+	if err != nil {
+		logrus.WithError(err).Fatal("failed go gather options")
+	}
+	if err := validateOptions(o); err != nil {
+		logrus.Fatalf("invalid options: %v", err)
+	}
+	level, _ := logrus.ParseLevel(o.logLevel)
+	logrus.SetLevel(level)
+
+	if err := addSchemes(); err != nil {
+		logrus.WithError(err).Fatal("failed to set up scheme")
+	}
+
+	var hiveClient ctrlruntimeclient.Client
+
+	kubeConfigs, _, err := util.LoadKubeConfigs(o.hiveKubeconfigPath, nil)
```

Comment:
The function to reload when kubeconfig changes should not be nil

Reply:
Updated.